lock/src/main/java/com/auth0/android/lock/Configuration.java

```diff
@@ -179,40 +183,76 @@ private Strategy filterStrategy(Strategy strategy, Set<String> connections) {
         return filtered;
     }
 
-    private List<Strategy> filterPasswordlessStrategies(List<Strategy> strategies, Set<String> connections) {
-        if (strategies == null || connections.isEmpty()) {
-            return strategies;
+    private Strategy filterPasswordlessStrategies(List<Strategy> strategies, Set<String> connections) {
+        if (strategies == null || strategies.isEmpty()) {
+            return null;
         }
-        List<Strategy> filtered = new ArrayList<>(strategies.size());
-        for (Strategy strategy : strategies) {
-            if (connections.contains(strategy.getName())) {
-                filtered.add(strategy);
+
+        if (connections.isEmpty()) {
```

Comment:
Also we are are not filtering the connections of the passwordless strategy, we are just returning it as it is, so if the client has two connections `conn1` & `conn2` inside lets say `Email` strategy, and wishes to use `conn2` it won't be possible since we are not filtering the first one